laws/profunctor.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const {identity, compose} = require('fantasy-combinators');
+const {dimap} = require('..');
+
+/**
+
+### Profunctor
+
+1. `p.dimap(f, g)` is equivalent to `p` (identity)
+2. `p.dimap(compose(f1)(f2)), compose(g1)(g2))` is equivalent to `p.dimap(f1, g1).dimap(f2, g2)` (composition)
+
+**/
+
+const identityʹ = t => eq => x => {
+    const a = t(x)[dimap](identity, identity);
+    const b = t(x);
+    return eq(a, b);
+};
+
+const composition = t => eq => x => {
+    const a = p[dimap](compose(identity)(identity), compose(identity)(identity));
+    const b = p[dimap](identity, identity)[dimap](identity, identity);
```

Comment:
Yes, yes they should.

On Fri, 29 Jan 2016, 22:22 Scott Christopher <notifications@github.com>
wrote:

> In laws/profunctor.js
> <https://github.com/fantasyland/fantasy-land/pull/124#discussion_r51324452>
> :
>
> > +### Profunctor
> > +
> > +1. `p.dimap(f, g)` is equivalent to `p` (identity)
> > +2. `p.dimap(compose(f1)(f2)), compose(g1)(g2))` is equivalent to `p.dimap(f1, g1).dimap(f2, g2)` (composition)
> > +
> > +**/
> > +
> > +const identityʹ = t => eq => x => {
> > +    const a = t(x)[dimap](identity, identity);
> > +    const b = t(x);
> > +    return eq(a, b);
> > +};
> > +
> > +const composition = t => eq => x => {
> > +    const a = p[dimap](compose(identity)(identity), compose(identity)(identity));
> > +    const b = p[dimap](identity, identity)[dimap](identity, identity);
>
> Should these lines use t(x)[dimap] rather than p[dimap] here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fantasyland/fantasy-land/pull/124/files#r51324452>.
>